packages/grid/data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -189,12 +189,51 @@ describe('<DataGrid /> - Filter', () => {
 
   describe('date operators', () => {
     [
-      { operator: 'is', value: new Date(2000, 11, 1), expected: ['12/1/2000'] },
-      { operator: 'not', value: new Date(2000, 11, 1), expected: ['1/1/2001', '1/1/2002'] },
-      { operator: 'after', value: new Date(2001, 0, 1), expected: ['1/1/2002'] },
-      { operator: 'onOrAfter', value: new Date(2001, 0, 1), expected: ['1/1/2001', '1/1/2002'] },
-      { operator: 'before', value: new Date(2001, 0, 1), expected: ['12/1/2000'] },
-      { operator: 'onOrBefore', value: new Date(2001, 0, 1), expected: ['12/1/2000', '1/1/2001'] },
+      { operator: 'is', value: '' },
+      { operator: 'is', value: undefined },
```

Comment:
This is weird for the `value` to take in undefined. When empty, shouldn't it be `null` if the type is rich, like a multi-select and an empty string if the type is simple?

Reply:
> This is weird for the value to take in undefined.

Using the `filterModel` prop it can be undefined.  

> When empty, shouldn't it be null if the type is rich, like a multi-select and an empty string if the type is simple?

Yes, and I would extend your suggestion saying that it should be typed as `any` and cast to the appropriated value: string for string columns, number for numeric columns and so on. We'll need to revisit this in #1320 to support an "one of" operator that allows to filter for rows that have one of the values specified.

Reply:
> Using the filterModel prop it can be undefined.

True, my point was that it doesn't seem to be something the types should allow.